app/src/main/java/com/itsaky/androidide/models/ProjectInfoDetails.kt

```diff
@@ -0,0 +1,52 @@
+/*
+ *  This file is part of AndroidIDE.
+ *
+ *  AndroidIDE is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  AndroidIDE is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *   along with AndroidIDE.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package com.itsaky.androidide.models
+
+import com.itsaky.androidide.data.projectInfo.ProjectInfo
+import com.itsaky.androidide.utils.TimeUtils
+import java.io.File
+
+class ProjectInfoDetails(
+  val name: String,
+  val file: File,
+  val cache: ProjectInfoCache
+)
+
+fun ProjectInfoDetails.toProjectInfo(): ProjectInfo = ProjectInfo(
+  name = name,
+  file = file,
+  cache = cache
+)
+
+fun ProjectInfo.toProjectInfoDetails(): ProjectInfoDetails = ProjectInfoDetails(
+  name = name,
+  file = file,
+  cache = cache
+)
```

Comment:
The presence of two classes, `ProjectInfo` and `ProjectInfoDetails`, that store similar data and are converted to each other may seem redundant, but this serves a specific design purpose of our code.

The reason for having two classes and the associated conversion functions is as follows:

1. Separation of Responsibilities: `ProjectInfo` can be considered a pure data model class that reflects the structure of data as it is stored in the database or transmitted through APIs. On the other hand, `ProjectInfoDetails` can be viewed as a class that models data in the context of the user interface or other parts of the application where design details are relevant. This separation of responsibilities can make code more organized and easier to maintain.

2. Flexibility: Having two separate classes gives us the flexibility to add fields or specific logic to each of them if necessary without affecting the other. This can be useful if specific requirements for one of the classes arise in the future.

3. Performance Improvements: In some cases, it may be more efficient to have a simplified data model class (such as `ProjectInfoDetails`) for certain operations in the user interface, reducing the overhead of unused data.

If you think this might be confusing, we might consider consolidating the classes if there is no clear benefit to keeping them separate.

Please let me know if you think we should continue with both classes or if you prefer a different approach.